### PR TITLE
Stop workers when controller stops

### PR DESCRIPTION
**Pull request by asymmetric**
This makes sure that all workers are terminated.

### CODE REVIEW

**Comment by zeenix** [comment]
why not? :) LGTM.

File: pkg/habitat/controller/controller.go [hunk 1]
```diff
@@ -757,10 +760,13 @@ func (hc *HabitatController) worker() {
 }
 
 func (hc *HabitatController) processNextItem() bool {
+	// Process an item, unless `queue.ShutDown()` has been called, in which case we exit.
 	key, quit := hc.queue.Get()
 	if quit {
 		return false
 	}
+
+	// Indicate to the queue that we've completed a unit of work.
```
Comment (LiliC) [inline pkg/habitat/controller/controller.go]: Not sure we need these comments, as `queue.Done(key)` is very descriptive.

File: pkg/habitat/controller/controller.go [hunk 2]
```diff
@@ -778,6 +784,7 @@ func (hc *HabitatController) processNextItem() bool {
 		return true
 	}
 
+	// If there was no error, tell the queue it can stop tracking failure history for the key.
```
Comment (LiliC) [inline pkg/habitat/controller/controller.go]: Same here as above, not sure this comment is needed.

File: pkg/habitat/controller/controller.go (hunk 1 above)
Comment (zeenix) [reply pkg/habitat/controller/controller.go]: Now that you mention it, yeah I also think these comments come close to what Linus calls over-documentation. :)

File: pkg/habitat/controller/controller.go (hunk 2 above)
Comment (asymmetric) [reply pkg/habitat/controller/controller.go]: I disagree about this one. It's not clear what `Forget` does exactly without having to dig through k8s' code, and the [`Forget`](https://github.com/kinvolk/habitat-operator/tree/3daa1e86d3e581c107044982de11507e9c896c3b/vendor/k8s.io/client-go/util/workqueue/default_rate_limiters.go#L117) method itself isn't documented.

**Pull request merged**